crypto/bio/b_sock.c

```diff
@@ -387,6 +387,8 @@ int BIO_socket_wait(int fd, int for_read, time_t max_time)
     struct timeval tv;
     time_t now;
 
+    if (fd >= FD_SETSIZE - 1)
```

Comment:
fd >= FD_SETSIZE but please in a separate PR.

Reply:
All right, will do in a separate PR.
Yet don't I need to check for `fd >= FD_SETSIZE - 1` since the function uses `fd + 1`?

Reply:
no.